pulpcore/cli/python/context.py

```diff
@@ -0,0 +1,91 @@
+from pulpcore.cli.common.context import (
+    EntityDefinition,
+    PulpEntityContext,
+    PulpRepositoryContext,
+    PulpRepositoryVersionContext,
+)
+
+
+class PulpPythonContentContext(PulpEntityContext):
+    ENTITY = "python content"
+    ENTITIES = "python content"
+    HREF = "python_python_package_content_href"
+    LIST_ID = "content_python_packages_list"
+    READ_ID = "content_python_packages_read"
+    CREATE_ID = "content_python_packages_create"
+
+
+class PulpPythonDistributionContext(PulpEntityContext):
+    ENTITY = "python distribution"
+    ENTITIES = "python distributions"
+    HREF = "python_python_distribution_href"
+    LIST_ID = "distributions_python_pypi_list"
+    READ_ID = "distributions_python_pypi_read"
+    CREATE_ID = "distributions_python_pypi_create"
+    UPDATE_ID = "distributions_python_pypi_partial_update"
+    DELETE_ID = "distributions_python_pypi_delete"
+
+    def preprocess_body(self, body: EntityDefinition) -> EntityDefinition:
+        body = super().preprocess_body(body)
+        for nullable in [
+            "publication",
+        ]:
+            if body.get(nullable) == "":
+                body[nullable] = None
+        return body
+
+
+class PulpPythonPublicationContext(PulpEntityContext):
+    ENTITY = "python publication"
+    ENTITIES = "python publications"
+    HREF = "python_python_publication_href"
+    LIST_ID = "publications_python_pypi_list"
+    READ_ID = "publications_python_pypi_read"
+    CREATE_ID = "publications_python_pypi_create"
+    DELETE_ID = "publications_python_pypi_delete"
+
+    def preprocess_body(self, body: EntityDefinition) -> EntityDefinition:
+        body = super().preprocess_body(body)
+        version = body.pop("version", None)
+        if version is not None:
+            repository_href = body.pop("repository")
+            body["repository_version"] = f"{repository_href}versions/{version}/"
+        return body
+
+
+class PulpPythonRemoteContext(PulpEntityContext):
+    ENTITY = "python remote"
+    ENTITIES = "python remotes"
+    HREF = "python_python_remote_href"
+    LIST_ID = "remotes_python_python_list"
+    CREATE_ID = "remotes_python_python_create"
+    BANDERSNATCH_ID = "remotes_python_python_from_bandersnatch"
```

Comment:
This could use some type annotation. `ClassVar[str]` i believe.
Also i'd expect a method on this class to use it.
Are we using it yet?

Reply:
```suggestion
    BANDERSNATCH_ID: ClassVar[str] = "remotes_python_python_from_bandersnatch"
```
And i see we aren't using it by now. So don't bother about that method just yet.